iOS_SDK/OneSignalSDK/OneSignalUser/Source/Requests/OSRequestAddAliases.swift

```diff
@@ -37,15 +38,16 @@ class OSRequestAddAliases: OneSignalRequest, OSUserRequest {
     var identityModel: OSIdentityModel
     let aliases: [String: String]
 
-    // requires a `onesignal_id` to send this request
-    func prepareForExecution() -> Bool {
-        if let onesignalId = identityModel.onesignalId, let appId = OneSignalConfigManager.getAppId() {
+    /// requires a `onesignal_id` to send this request
+    func prepareForExecution(newRecordsState: OSNewRecordsState) -> Bool {
```

Comment:
Why are we passing in this state as a parameter to prepareForExecution instead of the class's initializer? Do we expect that state object to change between calls to prepareForExecution?

Reply:
If the requests are uncached, they won't have the OSNewRecordsState.